fendermint/vm/interpreter/src/fvm/genesis/builtin.rs

```diff
@@ -0,0 +1,246 @@
+                None,
+            )
+            .context("failed to create system actor")?;
+
+        // Init actor
+        let (init_state, addr_to_id) = init::State::new(
+            state.store(),
+            genesis.chain_name.clone(),
+            &genesis.accounts,
+            &self.eth_builtin_ids,
+            self.eth_libs.len() as u64,
+        )
+        .context("failed to create init state")?;
+
+        state
+            .create_builtin_actor(
+                init::INIT_ACTOR_CODE_ID,
+                init::INIT_ACTOR_ID,
+                &init_state,
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create init actor")?;
+
+        // Cron actor
+        state
+            .create_builtin_actor(
+                cron::CRON_ACTOR_CODE_ID,
+                cron::CRON_ACTOR_ID,
+                &cron::State {
+                    entries: vec![], // TODO: Maybe with the IPC.
+                },
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create cron actor")?;
+
+        // Burnt funds actor (it's just an account).
+        state
+            .create_builtin_actor(
+                account::ACCOUNT_ACTOR_CODE_ID,
+                burntfunds::BURNT_FUNDS_ACTOR_ID,
+                &account::State {
+                    address: burntfunds::BURNT_FUNDS_ACTOR_ADDR,
+                },
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create burnt funds actor")?;
+
+        // A placeholder for the reward actor, beause I don't think
+        // using the one in the builtin actors library would be appropriate.
+        // This effectively burns the miner rewards. Better than panicking.
+        state
+            .create_builtin_actor(
+                account::ACCOUNT_ACTOR_CODE_ID,
+                reward::REWARD_ACTOR_ID,
+                &account::State {
+                    address: reward::REWARD_ACTOR_ADDR,
+                },
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create reward actor")?;
+
+        let eam_state = fendermint_actor_eam::State::new(
+            &state.store(),
+            genesis
+                .contract_deployers
+                .clone()
+                .into_iter()
+                .map(|a| a.0)
+                .collect(),
+        )?;
+        state
+            .create_actor_internal(
+                eam_code_cid,
+                eam::EAM_ACTOR_ID,
+                &eam_state,
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create EAM actor")?;
+
+        Ok(addr_to_id)
+    }
+
+    async fn replace_eam_actor<DB>(
+        &self,
+        state: &mut FvmGenesisState<DB>,
+        store: &MemoryBlockstore,
+        builtin: &mut Vec<CodeByName>,
+    ) -> anyhow::Result<(Cid, Cid)>
```

Comment:
We should rename this function or split it in two functions as its not clear what its doing by its name. We could then make it more general whereby the caller can send the actor name it wants to replace (and even a list in the future)